### DISCUSSION TITLE
Runnning with .jar file

### DISCUSSION THREAD

**Discussion by g00dtw0** [coobbi/Jre4android#10] [General]
I am in trouble to use jar file in jre4android. 
I can open the program using jre4android but I cannot open the data files(usually .xml format file) from the executed program (as attached and named BlastVeewer) because I could not see the data file and the key is not working. 
In window system, I can use this program(BlastVeewer) without any obstacles.
Could you let me know if I incorrectly use the jre4android for the purpose?

Thank you in advance. 
[blastveewer-5.5.2.zip](https://github.com/user-attachments/files/17809317/blastveewer-5.5.2.zip)

**Comment by coobbi** [comment]
Like this?
https://github.com/user-attachments/assets/312bd83d-d65a-4e9d-ab19-4ca6655229e1

I try jre4android pro, Add the xml file from setting page, then open the jar(BlastViewer), choose open file, you will find the xml file at classpath folder

@g00dtw0

**Comment by coobbi** [comment]
@g00dtw0 Or you can use this method:
https://github.com/user-attachments/assets/f68b134a-8127-4c05-a222-9ca8828ae4fd

**Reply by g00dtw0** [reply]
Thank you very much. I will try.

**Reply by g00dtw0** [reply]
https://github.com/user-attachments/assets/96b34acb-2bd1-4be9-97a1-60cdd58133bc


https://github.com/user-attachments/assets/8b9a2482-cb89-4fe5-8e0c-3dea825c6f7b

After add the xml file to classpath, I can not see the classpath folder which you showed in the program to open the xml file when I open the jar program. What do i have to see the classpath folder?

**Comment by coobbi** [comment]
you can try jre4android pro version
https://play.google.com/store/apps/details?id=com.coobbi.jre.pro

g00dtw0 ***@***.***> 于 2024年11月22日周五 07:33写道：

**Comment by coobbi** [comment]
free version will update after few months

coobbi ***@***.***> 于 2024年11月22日周五 08:45写道：

**Comment by g00dtw0** [comment]
Great, thank you for your reply and consideration on the update of free versioin. I will try paid version first.

**Reply by g00dtw0** [reply]
Coobbi, I have a same problem with your paid version. I can not see the classpath folder in the executed jar file.

**Comment by coobbi** [comment]
Can you take a video about problem and send it to me so I can see if I can
find the problem? My email address is: ***@***.***

g00dtw0 ***@***.***> 于2024年11月25日周一 10:58写道：

**Reply by g00dtw0** [reply]
Dear coobbi,

Thank you for your suggestion and discussion.
I got the classpath after few removal and reinstall of the paid version. Now I can open the data file from my jar program. 

By the way, I have one more challenge with another jar program as url linked below. 

https://www.jalview.org/downloads/jar/release/jalview-all-2.11.4.1-j1.8.jar

It seemed that I stucked in loading of the program in the paid version of the jre4android as you can see the movie below.

https://github.com/user-attachments/assets/88c2a73b-cb2c-49bf-be26-2aa373d6e13a


Great thanks in advance.

**Reply by g00dtw0** [reply]
Do I need to wait more time for opening the program?

**Reply by coobbi** [reply]
You can see the error message by running it from the command type.
![14871733358298_ pic](https://github.com/user-attachments/assets/d9fb643e-14ee-4e21-97fb-b46134ffd6d6)

According to the error message, jalview dynamically calls the Java program (java -xxx) when it starts. This method is not allowed on the Android platform, so it prompts permission denied.
